plugins/modules/instance.py

```diff
@@ -816,10 +816,9 @@ def _create_disk_register(self, **params: Any) -> None:
                 self.client, params.pop("stackscript_id")
             )
 
-        # Workaround for an edge case where a StackScript has no UDFs
-        # and the API rejects an empty list for stackscript_data
-        if len(params.get("stackscript_data") or {}) < 1:
-            params.pop("stackscript_data")
+        # StackScript data is expected to be specified as kwargs
+        if params.get("stackscript_data") is not None:
+            params.update(params.pop("stackscript_data"))
```

Comment:
I think it would be safer to check whether the type of `params["stackscript_data"]` is dict before passing it to the `update` function?